src/app_service/service/community_tokens/service.nim

```diff
@@ -286,34 +326,118 @@ QtObject:
 
-    let suggestedFees = decodeSuggestedFeesDto(responseJson["fees"])
-    let contractGasUnits = self.deployCollectiblesEstimate()
+  proc getTokensToBurn(self: Service, walletAndAmountList: seq[WalletAndAmount], contract: CommunityTokenDto): seq[Uint256] =
+    if contract.address == "":
+      error "Can't find contract"
+      return
+    let tokenOwners = self.getCommunityTokenOwners(contract.communityId, contract.chainId, contract.address)
+    let tokenIds = self.collectTokensToBurn(walletAndAmountList, tokenOwners)
+    if len(tokenIds) == 0:
+      error "Can't find token ids to burn"
+    return tokenIds
+
+  # TODO use temp fees for deployment also
+  proc buildTransactionFromTempFees(self: Service, addressFrom: string): TransactionDataDto =
+    return ens_utils.buildTransaction(parseAddress(addressFrom), 0.u256, $self.tempGasUnits,
+      if self.tempSuggestedFees.eip1559Enabled: "" else: $self.tempSuggestedFees.gasPrice, self.tempSuggestedFees.eip1559Enabled,
+      if self.tempSuggestedFees.eip1559Enabled: $self.tempSuggestedFees.maxPriorityFeePerGas else: "",
+      if self.tempSuggestedFees.eip1559Enabled: $self.tempSuggestedFees.maxFeePerGasM else: "")
+
+  proc selfDestructCollectibles*(self: Service, communityId: string, password: string, walletAndAmounts: seq[WalletAndAmount], contractUniqueKey: string) =
+    try:
+      let contract = self.findContractByUniqueId(contractUniqueKey)
+      let tokenIds = self.getTokensToBurn(walletAndAmounts, contract)
+      if len(tokenIds) == 0:
+        return
+      let addressFrom = self.contractOwner(contract.chainId, contract.address)
+      let txData = self.buildTransactionFromTempFees(addressFrom)
+      debug "Remote destruct collectibles ", chainId=contract.chainId, address=contract.address, tokens=tokenIds
+      let response = tokens_backend.remoteBurn(contract.chainId, contract.address, %txData, password, tokenIds)
+      let transactionHash = response.result.getStr()
+      debug "Remote destruct transaction hash ", transactionHash=transactionHash
+
+      var data = RemoteDestructArgs(communityToken: contract, transactionHash: transactionHash, status: TransactionStatus.Pending)
+      self.events.emit(SIGNAL_REMOTE_DESTRUCT_STATUS, data)
+
+      # observe transaction state
+      self.transactionService.watchTransaction(
+        transactionHash,
+        addressFrom,
+        contract.address,
+        $PendingTransactionTypeDto.CollectibleRemoteSelfDestruct,
+        $contract.toJsonNode(),
+        contract.chainId,
+      )
+    except Exception as e:
+      error "Remote self destruct error", msg = e.msg
+
+  proc computeSelfDestructFee*(self: Service, walletAndAmountList: seq[WalletAndAmount], contractUniqueKey: string) =
+    let contract = self.findContractByUniqueId(contractUniqueKey)
+    let tokenIds = self.getTokensToBurn(walletAndAmountList, contract)
+    if len(tokenIds) == 0:
+      warn "token list is empty"
+      return
```

Comment:
Can you put the part above in the `try`. It's actually the only part that could fail and cause a crash, because the threadpool start can't really fail.